crates/sui-rpc-api/src/client/mod.rs

```diff
@@ -393,3 +407,62 @@ fn status_from_error_with_metadata<T: Into<BoxError>>(err: T, metadata: Metadata
     *status.metadata_mut() = metadata;
     status
 }
+
+#[derive(Clone, Debug, Default)]
+pub struct AuthInterceptor {
+    auth: Option<tonic::metadata::MetadataValue<tonic::metadata::Ascii>>,
+}
+
+impl AuthInterceptor {
+    /// Enable HTTP basic authentication with a username and optional password.
+    pub fn basic<U, P>(username: U, password: Option<P>) -> Self
+    where
+        U: std::fmt::Display,
+        P: std::fmt::Display,
+    {
+        use base64::prelude::BASE64_STANDARD;
```

Comment:
I'm not sure but do we need base64 url encoding here?

Reply:
This is the required spec for basic auth with username/password. You can see how crates like reqwest handle this.